lib/Component.lua

```diff
@@ -424,4 +431,33 @@ function Component:_unmount()
 	self._handle = nil
 end
 
+--[[
+	Performs property validation, if it is enabled.
+]]
+function Component:_validateProps(props)
+	if not GlobalConfig.getValue("propertyValidation") then return end
+
+	local validator = self.validateProps
+
+	if validator == nil then return end
+	if typeof(validator) ~= "function" then
+		-- Hide as much as possible about error location, since this message
+		-- occurs from several possible call sites with different stack traces.
+		-- luacheck: ignore 6
+		error(("The value of validateProps must be a function (is a %q). Check the definition of the component extended at:\n%s"):format(
```

Comment:
Can we tweak this error message?

```lua
"The value of validateProps must be a function, but it is a %s." ..
"\nCheck the component defined at:\n%s"
```

Maybe this is a case to bring in `multilineError` or whatever I've been calling it lately!